File: 0881-boats-to-save-people/0881-boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    people.sort((a, b) => a - b)
    let i = 0, j = people.length-1, res = 0
    while (i <= j) {
        if (people[i] + people[j] <= limit) {
            i++
            j--
        } else {
            j--
        }
        res++
    }
    return res
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l, r = 0, len(people)-1
        num_boats = 0
        while l <= r:
            if people[l] + people[r] <= limit:
                l += 1
                r -= 1
            else:
                r -= 1
            num_boats += 1
        return num_boats